test/fuzz/parser_fuzzers.go

```diff
@@ -0,0 +1,337 @@
+// - version/(*Version).AtLeast(*Version)
+func FuzzParseVersion(data []byte) int {
+	f := fuzz.NewConsumer(data)
+	vString1, err := f.GetString()
+	if err != nil {
+		return 0
+	}
+	v1, err := version.ParseSemantic(vString1)
+	if err != nil {
+		return 0
+	}
+
+	// Test if the Version will crash (*Version).String()
+	_ = v1.String()
+
+	vString2, err := f.GetString()
+	if err != nil {
+		return 0
+	}
+	v2, err := version.ParseGeneric(vString2)
+	if err != nil {
+		return 0
+	}
+	_ = v1.AtLeast(v2)
+	return 1
+}
+
+// FuzzCreatePortAllocator implements a fuzzer
+// that targets:
+// - net.ParsePortRange(string(data)),
+// - portallocator.NewInMemory(*PortRange)
+// - portallocator./(*PortAllocator).AllocateNext()
+func FuzzCreatePortAllocator(data []byte) int {
+	pr, err := net.ParsePortRange(string(data))
+	if err != nil {
+		return 0
+	}
+	r, err := portallocator.NewInMemory(*pr)
+	if err != nil {
+		return 0
+	}
+
+	// Test if the Portrange will crash Allocatenext
+	for i := 0; i < 20; i++ {
+		_, err = r.AllocateNext()
+		if err != nil {
+			return 0
+		}
+	}
+	return 1
+}
+
+// FuzzUrlsMatch implements a fuzzer
+// that targets credentialprovider.URLsMatchStr
+func FuzzUrlsMatch(data []byte) int {
+	f := fuzz.NewConsumer(data)
+	glob, err := f.GetString()
+	if err != nil {
+		return 0
+	}
+	target, err := f.GetString()
+	if err != nil {
+		return 0
+	}
+	_, _ = credentialprovider.URLsMatchStr(glob, target)
+	return 1
+}
+
+// FuzzParseEnv implements a fuzzer
+// that targets envutil.ParseEnv
+func FuzzParseEnv(data []byte) int {
+	f := fuzz.NewConsumer(data)
+
+	// Create a pseudo-random spec.
+	// Will be used as argument to the fuzz target
+
+	// length of slice:
+	qty, err := f.GetInt()
+	if err != nil {
+		return 0
+	}
+	ss := make([]string, qty, qty)
+
+	// fill slice with values
+	for i := 0; i < qty; i++ {
+		s, err := f.GetString()
+		if err != nil {
+			return 0
+		}
+		ss = append(ss, s)
+	}
+	var r io.Reader
+	_, _, _, _ = envutil.ParseEnv(ss, r)
+	return 1
+}
+
+// FuzzParseQOSReserve implements a fuzzer
+// that targets cm.ParseQOSReserved
+func FuzzParseQOSReserve(data []byte) int {
+	f := fuzz.NewConsumer(data)
+
+	// Create a pseudo-random map.
+	// Will be used as argument to the fuzz target
+	m := make(map[string]string)
+	err := f.FuzzMap(&m)
+	if err != nil {
+		return 0
+	}
+	_, _ = cm.ParseQOSReserved(m)
+	return 1
+}
+
+// FuzzParseClientGenTags implements a fuzzer
+// that targets util.ParseClientGenTags
+func FuzzParseClientGenTags(data []byte) int {
+	f := fuzz.NewConsumer(data)
+
+	// Create a pseudo-random lines
+	// Will be used as arguemnt to the fuzz target
+
+	// length of slice:
+	qty, err := f.GetInt()
+	if err != nil {
+		return 0
+	}
+	lines := make([]string, qty, qty)
+
+	// fill slice with values:
+	for i := 0; i < qty; i++ {
+		line, err := f.GetString()
+		if err != nil {
+			return 0
+		}
+		lines = append(lines, line)
+	}
+	_, _ = util.ParseClientGenTags(lines)
+	return 1
+}
```

Comment:
This is not a good fuzzing target. Input is trusted.

Reply:
Removed